sleap/nn/utils.py

```diff
@@ -126,3 +126,52 @@ def match_points(points1: tf.Tensor, points2: tf.Tensor) -> Tuple[tf.Tensor, tf.
         axis=-1,
     )
     return tf_linear_sum_assignment(dists)
+
+
+def reset_input_layer(
+    keras_model: tf.keras.Model,
+    new_shape: Optional[Tuple[Optional[int], Optional[int], Optional[int], int]] = None,
+):
+    """Returns a copy of `keras_model` with input shape reset to `new_shape`.
+
+    This method was modeified from https://stackoverflow.com/a/58485055.
+
+    Args:
+        keras_model: `tf.keras.Model` to return a copy of (with input shape reset).
+        new_shape: Shape of the returned model's input layer.
+
+    Returns:
+        A copy of `keras_model` with input shape `new_shape`.
+    """
+
+    if new_shape is None:
+        new_shape = (None, None, None, keras_model.input_shape[-1])
+
+    model_config = keras_model.get_config()
+
+    input_layer_name = model_config["layers"][0]["name"]
+    model_config["layers"][0] = {
+        "name": f"{input_layer_name}",
+        "class_name": "InputLayer",
+        "config": {
+            "batch_input_shape": new_shape,
+            "dtype": "float32",
+            "sparse": False,
+            "name": f"{input_layer_name}",
+        },
+        "inbound_nodes": [],
+    }
+
+    model_config["layers"][1]["inbound_nodes"] = [[[f"{input_layer_name}", 0, 0, {}]]]
+    model_config["input_layers"] = [[f"{input_layer_name}", 0, 0]]
+
+    new_model: tf.keras.Model = keras_model.__class__.from_config(
+        model_config, custom_objects={}
+    )  # Change custom objects if necessary
+
+    # Iterate over all the layers that we want to get weights from
+    weights = [layer.get_weights() for layer in keras_model.layers[1:]]
+    for layer, weight in zip(new_model.layers[1:], weights):
+        layer.set_weights(weight)
```

Comment:
Is it perhaps less general to assume that the input layer is `keras_model.layers[0]`?

I guess this is baked in above as well, but why do we skip `layers[0]`? In general it should not have weights anyway, but I'm not seeing what's the reason to skip it.